Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -624,6 +624,35 @@ lemma nonsingular_add {x₁ x₂ y₁ y₂ : F} (h₁ : W.Nonsingular x₁ y₁)
   nonsingular_neg <| nonsingular_negAdd h₁ h₂ hxy
 #align weierstrass_curve.nonsingular_add WeierstrassCurve.Affine.nonsingular_add
 
+variable {x₁ x₂ : F} (y₁ y₂ : F) (hx : x₁ ≠ x₂)
+
+/-- The formula $x(P_1 + P_2) = x(P_1 - P_2) - \psi(P_1)\psi(P_2) / (x(P_2) - x(P_1)) ^ 2$,
+where $\psi(x,y) = 2y + a_1 x + a_3$. -/
+lemma addX_eq_subX_sub :
+    W.addX x₁ x₂ (W.slope x₁ x₂ y₁ y₂) = W.addX x₁ x₂ (W.slope x₁ x₂ y₁ (W.negY x₂ y₂))
+      - (y₁ - W.negY x₁ y₁) * (y₂ - W.negY x₂ y₂) / (x₂ - x₁) ^ 2 := by
+  simp_rw [slope, if_neg hx, addX, negY, ← neg_sub x₁, neg_sq]
+  field_simp [sub_ne_zero.mpr hx]; ring
+
+/-- The formula $y(P_1)(x(P_2)-x(P_3)) + y(P_2)(x(P_3)-x(P_1)) + y(P_3)(x(P_1)-x(P_2)) = 0$,
+assuming that $P_1 + P_2 + P_3 = O$. -/
+lemma cyclic_sum_Y_mul_X_sub_X :
+    letI x₃ := W.addX x₁ x₂ (W.slope x₁ x₂ y₁ y₂)
```

Comment:
`let`?

Reply:
I think I do want to inline this expression; it's just it appears multiple times so I'd like a shorthand. Without `letI` maybe there'll be issues rewriting with it, I'm not sure.

Reply:
Thanks for testing! Removed the `I`